Sources/SparkConnect/SparkConnectClient.swift

```diff
@@ -0,0 +1,228 @@
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import Foundation
+import GRPCCore
+import GRPCNIOTransportHTTP2
+import GRPCProtobuf
+import Synchronization
+
+/// Conceptually the remote spark session that communicates with the server
+public actor SparkConnectClient {
+  let clientType: String = "swift"
+  let url: URL
+  let host: String
+  let port: Int
+  let userContext: UserContext
+  var sessionID: String? = nil
+
+  /// Create a client to use GRPCClient.
+  /// - Parameters:
+  ///   - remote: A string to connect `Spark Connect` server.
+  ///   - user: A string for the user ID of this connection.
+  init(remote: String, user: String) {
+    self.url = URL(string: remote)!
+    self.host = url.host() ?? "localhost"
+    self.port = self.url.port ?? 15002
+    self.userContext = user.toUserContext
+  }
+
+  /// Stop the connection. Currently, this API is no-op because we don't reuse the connection yet.
+  func stop() {
+  }
+
+  /// Connect to the `Spark Connect` server with the given session ID string.
+  /// As a test connection, this sends the server `SparkVersion` request.
+  /// - Parameter sessionID: A string for the session ID.
+  /// - Returns: An `AnalyzePlanResponse` instance for `SparkVersion`
+  func connect(_ sessionID: String) async throws -> AnalyzePlanResponse {
+    try await withGRPCClient(
+      transport: .http2NIOPosix(
+        target: .dns(host: self.host, port: self.port),
+        transportSecurity: .plaintext
+      )
+    ) { client in
+      // To prevent server-side `INVALID_HANDLE.FORMAT (SQLSTATE: HY000)` exception.
+      if UUID(uuidString: sessionID) == nil {
+        throw SparkConnectError.InvalidSessionIDException
+      }
+
+      self.sessionID = sessionID
+      let service = SparkConnectService.Client(wrapping: client)
+      let version = AnalyzePlanRequest.SparkVersion()
+      var request = AnalyzePlanRequest()
+      request.clientType = clientType
+      request.userContext = userContext
+      request.sessionID = sessionID
+      request.analyze = .sparkVersion(version)
+      let response = try await service.analyzePlan(request)
+      return response
+    }
+  }
+
+  /// Create a ``ConfigRequest`` instance for `Set` operation.
+  /// - Parameter map: A map of key-value string pairs.
+  /// - Returns: A ``ConfigRequest`` instance.
+  func getConfigRequestSet(map: [String: String]) -> ConfigRequest {
+    var request = ConfigRequest()
+    request.operation = ConfigRequest.Operation()
+    var set = ConfigRequest.Set()
+    set.pairs = map.toSparkConnectKeyValue
+    request.operation.opType = .set(set)
+    return request
+  }
+
+  /// Request the server to set a map of configurations for this session.
+  /// - Parameter map: A map of key-value pairs to set.
+  /// - Returns: Always return true.
+  func setConf(map: [String: String]) async throws -> Bool {
```

Comment:
I wonder why don't return `ConfigResponse` like [Scala](https://github.com/apache/spark/blob/master/sql/connect/common/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala#L160) but a `Bool`?

Reply:
This is a different implementation choice to encapsulate those details.

Technically, each language client maintains it's `SparkConnectClient` in private scope like `private[sql]`.